sentry/src/main/java/io/sentry/ICpuCollector.java

```diff
@@ -0,0 +1,15 @@
+package io.sentry;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+/** Used for collecting data about cpu load when a transaction is active. */
+@ApiStatus.Internal
+public interface ICpuCollector {
```

Comment:
`m`: just thinking, what if we make a generic `interface ICollector<T>` instead and make `setup` method for MemoryCollector a no-op? Then we could have a list of `ICollector<T>` in SentryOptions instead of two separate property for memory and cpu (and this is also extendable to other collectors in the future).

Reply:
yeah, I thought about it, too
agree, let's do it